File: app/src/main/java/org/rm3l/ddwrt/service/tasks/RouterModelUpdaterServiceTask.java
package org.rm3l.ddwrt.service.tasks;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;

import org.rm3l.ddwrt.resources.conn.NVRAMInfo;
import org.rm3l.ddwrt.resources.conn.Router;
import org.rm3l.ddwrt.utils.SSHUtils;
import org.rm3l.ddwrt.utils.Utils;

/**
 * Created by rm3l on 30/07/15.
 */
public class RouterModelUpdaterServiceTask extends AbstractBackgroundServiceTask {

    public static final String DEFAULT_VALUE = "-";

    public RouterModelUpdaterServiceTask(@NonNull Context ctx) {
        super(ctx);
    }

    @Override
    public void runBackgroundServiceTask(@NonNull Router router) throws Exception {

        final SharedPreferences routerPreferences = mCtx.getSharedPreferences(
                router.getUuid(), Context.MODE_PRIVATE);

        if (routerPreferences == null) {
            return;
        }

        final NVRAMInfo nvramInfo = SSHUtils.getNVRamInfoFromRouter(
                mCtx,
                router,
                globalPreferences,
                NVRAMInfo.MODEL);

        if (nvramInfo == null) {
            return;
        }

        final String routerModel = nvramInfo.getProperty(NVRAMInfo.MODEL, DEFAULT_VALUE);

        final String routerModelFromPrefs =
                routerPreferences.getString(NVRAMInfo.MODEL, DEFAULT_VALUE);
        //noinspection ConstantConditions
        if (!(DEFAULT_VALUE.equals(routerModel) ||
                routerModelFromPrefs.equals(routerModel))) {
            routerPreferences
                    .edit()
                    .putString(NVRAMInfo.MODEL, routerModel)
                    .apply();
            Utils.requestBackup(mCtx);
        }
    }
}

File: app/src/main/java/org/rm3l/router_companion/service/tasks/RouterModelUpdaterServiceTask.kt
package org.rm3l.router_companion.service.tasks

import android.content.Context
import org.rm3l.router_companion.resources.conn.NVRAMInfo
import org.rm3l.router_companion.resources.conn.Router
import org.rm3l.router_companion.utils.SSHUtils
import org.rm3l.router_companion.utils.Utils

/**
 * Created by rm3l on 30/07/15.
 */
class RouterModelUpdaterServiceTask(ctx: Context) : AbstractBackgroundServiceTask(ctx) {

  @Throws(Exception::class)
  override fun runBackgroundServiceTask(router: Router) {

    val routerPreferences = mCtx.getSharedPreferences(router.templateUuidOrUuid, Context.MODE_PRIVATE) ?: return

    val nvramInfo = SSHUtils.getNVRamInfoFromRouter(mCtx, router, globalPreferences,
        NVRAMInfo.MODEL) ?: return

    val routerModel = nvramInfo.getProperty(NVRAMInfo.MODEL, DEFAULT_VALUE)

    val routerModelFromPrefs = routerPreferences.getString(NVRAMInfo.MODEL, DEFAULT_VALUE)

    if (!(DEFAULT_VALUE == routerModel || routerModelFromPrefs == routerModel)) {
      routerPreferences.edit().putString(NVRAMInfo.MODEL, routerModel).apply()
      Utils.requestBackup(mCtx)
    }
  }

  companion object {
    val DEFAULT_VALUE = "-"
  }
}
